src/gas_estimator.rs

```diff
@@ -60,6 +67,21 @@ impl GasTracker {
         }
     }
 
+    /// Increases the history size limit
+    /// returns an error if the history is already larger than the input size
```

Comment:
maybe add a way to get a reference to the history? to perform custom operations

Not sure this needs it's own error type either if it's already larger you can just do nothing? A function to get the current size might be useful here.

Reply:
Added a public history fetching method